src/redis-benchmark.c

```diff
@@ -1424,21 +1420,24 @@ int parseOptions(int argc, char **argv) {
             config.keepalive = atoi(argv[++i]);
         } else if (!strcmp(argv[i],"-h")) {
             if (lastarg) goto invalid;
-            config.hostip = strdup(argv[++i]);
+            config.conn_info.hostip = strdup(argv[++i]);
```

Comment:
Note that multiple hostip's might be configured so you need to free the previous one before this assignment. But there's a mixup here because the default value isn't dynamically allocated so you can't free it. And it gets even more mixed up because the value set in the parse url function is and `sds` and not a simple `char*`. You need to sort all this out they it's done in redis-cli. 
This is also true for all the other strings in the config and even, perhaps the generated `dbnumstr`.